core/src/main/java/org/elasticsearch/cluster/routing/ShardRouting.java

```diff
@@ -274,6 +274,10 @@ public boolean allocatedPostIndexCreate(IndexMetaData indexMetaData) {
         return true;
     }
 
+    public boolean isPeerRecovery() {
+        return !primary() || relocatingNodeId() != null;
```

Comment:
can we change to use isRelocationTarget() instead of relocatingNodeId() != null ? I think it's confusing that a relocation source will return true here.